app.go

```diff
@@ -483,8 +483,17 @@ func (app *App) provide(constructor interface{}) {
 	}
 
 	if a, ok := constructor.(Annotated); ok {
-		if err := app.container.Provide(a.Target, dig.Name(a.Name)); err != nil {
-			app.err = err
+		switch {
+		case len(a.Group) > 0 && len(a.Name) > 0:
+			app.err = fmt.Errorf("fx.Annotate may not specify both name and group for %v", constructor)
+		case len(a.Name) > 0:
+			if err := app.container.Provide(a.Target, dig.Name(a.Name)); err != nil {
+				app.err = err
+			}
+		default:
+			if err := app.container.Provide(a.Target, dig.Group(a.Group)); err != nil {
```

Comment:
This should explicitly check for `len(a.Group) > 0` because using fx.Annotated without any annotations (`fx.Annotated{Target: fn}`) is valid usage/equivalent to not having fx.Annotated.